### PR TITLE
Use unbuild package for publishing

### PR DESCRIPTION
**Pull request by flashdesignory**
This pr changes how we build the package. 
It's using [unbuild](https://www.npmjs.com/package/unbuild) to ensure esm and cjs files are build correctly.

Verified with:
- [x] Nuxt TPC
- [ ] Next TPC

@kara

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,28 +2,21 @@
   "name": "third-party-capital",
   "version": "1.0.26",
   "description": "",
-  "main": "./lib/cjs/index.js",
-  "types": "./lib/cjs/types/index.d.ts",
-  "files": [
-    "lib/**/*"
-  ],
+  "type": "commonjs",
```
Comment (harlan-zw) [inline package.json:5]: I'd recommend exporting as a module by default

File: package.json [hunk 2]
```diff
@@ -2,28 +2,21 @@
...
-      "import": {
-        "types": "./lib/esm/types/index.d.ts",
-        "default": "./lib/esm/index.js"
-      },
-      "require": {
-        "types": "./lib/cjs/types/index.d.ts",
-        "default": "./lib/cjs/index.js"
-      }
+      "import": "./dist/index.mjs",
+      "require": "./dist/index.cjs"
```
Comment (harlan-zw) [inline package.json:9]: you can provide a top level `types` option and a `types` here to typescript support, see https://github.com/nuxt/nuxt/blob/main/packages/nuxt/package.json#L20

File: package.json (hunk 1 above)
Comment (flashdesignory) [reply package.json]: worked after switching jest.config.js to jest.config.ts

File: package.json (hunk 2 above)
Comment (flashdesignory) [reply package.json]: cool - adjusted the export to below:

```
"exports": {
    ".": {
      "import": "./dist/index.mjs",
      "require": "./dist/index.cjs",
      "types": "./dist/index.d.mts"
    }
  },
```

**Review by housseindjirdeh — APPROVED** [review]

**Pull request merged**